### PR TITLE
Allow to reduce size of e3sm log, by optionally suppressing pre/post condition checks warnings

### PR DESCRIPTION
**Pull request by bartgol**
A couple of mods:

- Change construction of pre/post condition checks: do not pass CheckFailHandling as template arg. It's pointless, it can be a runtime arg.
- Each atm proc can receive the parameter `repair_log_level`, which establishes the level of logging for when the pre/post condition checks fail, and are repaired. Default is 'warn' for standalone, and 'trace' for v1 runs. This means that, with likely choices of atm log level, standalone will print the warnings, while v1 will not, keeping logs sizes under control.

### CODE REVIEW

**Review by jgfouca — DISMISSED** [review]

File: components/scream/src/share/atm_process/atmosphere_process.hpp [hunk 1]
```diff
@@ -220,20 +220,21 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
...
-  template<typename FPC, CheckFailHandling CFH = CheckFailHandling::Fatal, typename... Args>
+  template<typename FPC, typename... Args>
   void add_precondition_check (const Args... args);
-  template<typename FPC, CheckFailHandling CFH = CheckFailHandling::Fatal, typename... Args>
+  template<typename FPC, typename... Args>
   void add_postcondition_check (const Args... args);
-  template<typename FPC, CheckFailHandling CFH = CheckFailHandling::Fatal, typename... Args>
+  template<typename FPC, typename... Args>
   void add_invariant_check (const Args... args);
+  template<typename FPC, typename... Args>
```
Comment (tcclevenger) [inline components/scream/src/share/atm_process/atmosphere_process.hpp:237]: This line should not be here.

File: components/scream/src/share/atm_process/atmosphere_process.cpp [hunk 2]
```diff
@@ -450,6 +468,49 @@ get_internal_field(const std::string& field_name) const {
...
+  // update a field, without that appearing in the dag.
+  for (const auto& ptr : pc->repairable_fields()) {
+    const auto& fid = ptr->get_header().get_identifier();
+    EKAT_REQUIRE_MSG (
+        has_computed_field(fid) || has_computed_group(fid.name(),fid.get_grid_name()),
+        "Error! Input property check can repair a non-computed field.\n"
+        "  - Atmosphere process name: " + name() + "\n"
+        "  - Property check name: " + name() + "\n");
+  }
+  m_precondition_checks.push_back(std::make_pair(pc,cfh));
```
Comment (tcclevenger) [inline components/scream/src/share/atm_process/atmosphere_process.cpp:493]: This pair (and one below) is backwards. Should be `(cfh,pc)`. Maybe the structure of `m_precondition_checks` and `add_precondition_check (pc, cfh)` should match.

File: components/scream/src/share/atm_process/atmosphere_process.cpp [hunk 3]
```diff
@@ -450,6 +468,49 @@ get_internal_field(const std::string& field_name) const {
...
+  // update a field, without that appearing in the dag.
+  for (const auto& ptr : pc->repairable_fields()) {
+    const auto& fid = ptr->get_header().get_identifier();
+    EKAT_REQUIRE_MSG (
+        has_computed_field(fid) || has_computed_group(fid.name(),fid.get_grid_name()),
+        "Error! Input property check can repair a non-computed field.\n"
+        "  - Atmosphere process name: " + name() + "\n"
+        "  - Property check name: " + name() + "\n");
+  }
+  m_postcondition_checks.push_back(std::make_pair(pc,cfh));
```
Comment (tcclevenger) [inline components/scream/src/share/atm_process/atmosphere_process.cpp:511]: Same as above.

**Review by tcclevenger — COMMENTED** [review]
The only other thing keeping me from compiling is line 540-544 in `src/share/tests/atm_process_test.cpp` need to be updated (remove the template and add as 2nd arg).

**Review by jgfouca — APPROVED** [review]

**Pull request merged**